experiments/models.py

```diff
@@ -196,10 +196,19 @@ class ParticipantSession(models.Model):
     )
     experiment_state = models.PositiveIntegerField(null=True)
 
-    group = models.ForeignKey('TargetGroup', on_delete=models.PROTECT)
+    group = models.ForeignKey(
+        'TargetGroup',
+        on_delete=models.PROTECT,
+        null=True
```

Comment:
does `null=True` mean you are going to use sessions even when no groups are configured?

Reply:
Yes, I wanted to piggy back on the session, to make the ParticipantSession to decided what the subject_id for an participant is in a given session, whether there are groups or not.
Sometimes, i feel it might be a wise decision to create an default group for an Experiment with a default group_name. Than we can use the default name for jspsych experiment with only one group. For within subject designs e.g.

Reply:
Do you see any problems with this desicion that I should fix?

Reply:
Maybe, yes;

However, I don't think serialization would actually break as there's no instance where a `ParticipantSession` without a group will be serialized?

Reply:
Adding default groups as follows in a post_save signal in experiment/signals.py, breaks pretty much everything in /api/tests.py/ :-)...

Pretty, much all test are fixed when I remove the default group in api.test.TestTargetGroupAllocation.setupTestData()
```python
class TestTargetGroupAllocation(APITestCase):
    @classmethod
    def setUpTestData(cls):
        access_id = uuid.uuid4()
        cls.exp = Experiment.objects.create(
            access_id=access_id,
            state=Experiment.OPEN,
            approved=True
        )
        # Remove the "default" group as that feature is newer that this tests"
        cls.exp.targetgroup_set.all().delete()
        # And Run the test as The tests were designed
        cls.group_a = cls.exp.targetgroup_set.create(name='A', completion_target=2)
        cls.group_b = cls.exp.targetgroup_set.create(name='B', completion_target=2)
```

However, than there remain other issues. The BaseUploadView is than also broken:
Because it contains the following code:
```python
        if self.experiment.has_groups():
            # If the experiment is configured to use target groups,
            # then session ids are mandatory
            raise ConfigError(code=ResultCodes.ERR_NO_SESSION,
                              detail='Missing participant session id')
```
Since doing `Experiment.save()` restores a default group if there isn't any. This would make it impossible to use this view, since every Experiment will alway contain at least one session.
The `is self.experiment.has_groups()` could be relaxed to `if self.experiment.targetgroups_set.count() > 1` Would any of you know what would be wise?

Reply:
> The is self.experiment.has_groups() could be relaxed to if self.experiment.targetgroups_set.count() > 1 Would any of you know what would be wise?

I think that would be a good solution. The reason we put an error here is to avoid a situation where the user intends to use target groups but is using the wrong jspsych code.

Reply:
I agree, it should be fine to relax it. 

Maybe rename `has_groups` to `must_uses_sessions` or `has_multiple_groups` (or something similar) would make it less confusing, as we'd be able to specify more clearly that one should use the session API if using multiple groups

Reply:
So this issue is now fixed as well in af3043db1a18cda938cc73776837ee1cb73e3000 . Since an experiment will always have groups, I changed the Experiment.has_groups to Experiment.uses_groups. Since we consider an experiment to be designed to use groups when it has more than 1 group.